src/system/sys_profiler.hh

```diff
@@ -369,16 +390,76 @@ protected:
     int child_timers[max_n_childs];
 
     /**
-     * Total number of bytes allocated directly in this frame (not include subframes).
+     * Total number of bytes allocated in this frame. After 
+     * Profiler::propagate_timers call will also contain values from children.
      */
     size_t total_allocated_;
     /**
-     * Total number of bytes deallocated directly in this frame (not include subframes).
+     * Total number of bytes deallocated in this frame. After 
+     * Profiler::propagate_timers call, will also contain values from children.
      */
     size_t total_deallocated_;
-
-
+    /**
+     * Maximum number of bytes allocated at one time in this frame. After 
+     * Profiler::propagate_timers call, maximum value will be taken from this
+     * Timer and also from all children Timers.
+     */
+    size_t max_allocated_;
+    /**
+     * Current number of bytes allocated in this frame at the same time. 
+     * This value is used to monitor maximum bytes allocated. When notify_free
+     * and notify_malloc is called this values is changed and new maximum 
+     * is tested.
+     */
+    size_t current_allocated_;
+    
+    /**
+     * Number of times new/new[] operator was used in this scope
+     */
+    int alloc_called;
+    /**
+     * Number of times delete/delete[] operator was used in this scope
+     */
+    int dealloc_called;
+    
+    #ifdef FLOW123D_HAVE_PETSC
+    /**
+     * Number of bytes used by Petsc at the start of time-frame
+     */
+    PetscLogDouble petsc_start_memory;
+    /**
+     * Number of bytes used by Petsc at the end of time-frame
+     */
+    PetscLogDouble petsc_end_memory;
+    /**
+     * Difference between start and end of a petsc memory usage 
+     */
+    PetscLogDouble petsc_memory_difference;
+    /**
+     * Maximum amount of memory used that was PetscMalloc()ed at any time 
+     * during this run.
+     *
+     * The memory usage reported here includes all Fortran arrays (that may be
+     * used in application-defined sections of code).
+     */
+    PetscLogDouble petsc_peak_memory;
+    /**
+     * Local maximum amount of memory used that was PetscMalloc()ed 
+     * used during time-frame pause/resume. Auxilary variable for storing 
+     * local memory used when pause is called.
+     */
+    PetscLogDouble petsc_local_peak_memory;
+    #endif // FLOW123D_HAVE_PETSC
+    
     friend class Profiler;
+    friend std::ostream & operator <<(std::ostream&, const Timer&);
+    
+    /**
+     * if under unit testing, specify friend so protected members can be tested
+     */
+    #ifdef __UNIT_TEST__
```

Comment:
It is still necessary. In testing we need to use `friend` pattern. If we were to use inheritance we would still not able to access protected members of Profiler since they belong to static instance of Profiler.

Right now, we set `ProfilerTest` as friend of `Profiler` and we inherit `Gtest` also in `ProfilerTest` class. This way we can access all protected members as well as create tests.